djaopsp/views/content.py

```diff
@@ -2,9 +2,30 @@
 # see LICENSE.
 
 from pages.views.elements import PageElementView, PageElementEditableView
+from pages.views.sequences import (SequenceProgressView as BaseSequenceProgressView, 
+    SequencePageElementView as BaseSequencePageElementView)
 
 from ..mixins import AccountMixin
+from ..models import SurveyEvent
 
+class SequenceProgressView(BaseSequenceProgressView):
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        elements = context.get('elements', [])
+
+        for element in elements:
+            element.is_survey_event = SurveyEvent.objects.filter(element=element.page_element).exists()
+        
+        context['elements'] = elements
+        
+        return context
+
+class SequencePageElementView(BaseSequencePageElementView):
+
+    def get_object(self, queryset=None):
+        element = super().get_object(queryset)
+        element.is_survey_event = SurveyEvent.objects.filter(element=element.page_element).exists()
```

Comment:
has commented in `djaodjin-pages`, defining a `decorate_element` method in a common mixin might be beneficial.